app/helpers/users_helper.rb

```diff
@@ -61,10 +61,13 @@ def rtl_safe_username(user)
 
   def user_link(user, url_opts = nil, **link_opts)
     url_opts ||= {}
-    if deleted_user?(user)
-      link_to 'deleted user', '#', { dir: 'ltr' }.merge(link_opts)
+    anchortext = link_opts[:anchortext]
+    if !anchortext.nil?
+      link_to anchortext, user_url(user, **url_opts), { dir: 'ltr' }.merge(link_opts)
+    elsif deleted_user?(user)
+      link_to 'deleted user', '#', { dir: 'ltr' }.merge(link_opts).except(:anchortext)
     else
-      link_to user.rtl_safe_username, user_url(user, **url_opts), { dir: 'ltr' }.merge(link_opts)
+      link_to user.rtl_safe_username, user_url(user, **url_opts), { dir: 'ltr' }.merge(link_opts).except(:anchortext)
```

Comment:
Optional: let's DRY this a little and extract `{ dir: 'ltr' }.merge(link_opts).except(:anchortext)` into a variable available for all branches and pass it instead.